firedrake/mesh.py

```diff
@@ -1542,6 +1542,48 @@ def cell_parent_cell_map(self):
         return op2.Map(self.cell_set, self._parent_mesh.cell_set, 1,
                        self.cell_parent_cell_list, "cell_parent_cell")
 
+    @property
+    def cell_parent_base_cell_list(self):
+        """Return a list of parent mesh base cells numbers in vertex only
+        mesh cell order.
+        """
+        if not isinstance(self._parent_mesh, ExtrudedMeshTopology):
+            raise AttributeError("Parent mesh is not extruded")
+        cell_parent_base_cell_list = np.copy(self.topology_dm.getField("parentcellbasenum"))
+        self.topology_dm.restoreField("parentcellbasenum")
+        return cell_parent_base_cell_list
+
+    @property
+    def cell_parent_base_cell_map(self):
```

Comment:
Should these be cached properties? They look quite expensive to compute.

Reply:
Interesting that `pyop2.utils.cached_property` doesn't use any of the code in `pyop2.caching`.

Reply:
For now I'll just use `pyop2.utils.cached_property` and add a todo note for when I have moving meshes